error.go

```diff
@@ -55,12 +58,19 @@ func IsGuestAPINotAvailable(err error) bool {
 	if strings.HasSuffix(c.Error(), guestNamespaceCreationErrorSuffix) {
 		return true
 	}
-	matched, matchErr := regexp.MatchString(guestDNSNotReadyPattern, c.Error())
-	if matchErr != nil {
-		return false
+
+	patterns := []string{
+		guestDNSNotReadyPattern,
+		guestTransientInvalidCertificatePattern,
 	}
-	if matched {
-		return true
+	for _, pattern := range patterns {
+		matched, matchErr := regexp.MatchString(pattern, c.Error())
+		if matchErr != nil {
+			return false
+		}
```

Comment:
How can `matchErr` occur? Is this maybe just a compilation error? In this case we should compile the regexp on program initialization and just use it more safely here at runtime. That would be super cool.

Reply:
Good point, done PTAL.